### PR TITLE
Some post tutorial updates

### PR DESCRIPTION
**Pull request by cdonnay**
1. Change the way the `condense_ballots()` method works in profiles. Rather than altering the original profile, it returns a new profile. This gives users the option to preserve the original profile.
2. Add an IRV election class, which is just a wrapper for STV with 1 seat.
3. Add default option to Borda election class, so users do not have to input a score vector if they want to use the traditional Borda vector.
4. Alter STV class so that the remaining candidates are always listed in order of current first place votes.
5. Add print statement to `BallotGraph` so that when you draw the graph without labels, it prints a dictionary of candidate labels for you.
6. Add several methods to `PairwiseComparisonGraph`. Added two boolean methods that return True if there is a condorcet winner or if there is a condorcet cycle. Added two get methods that return the winner or the cycles. Cached the results of `dominating_tiers` and `get_condorcet_cycles`.
7. Fixed an error in the `PreferenceProfile` tail method.
8. Added optional `to_float` method to `first_place_votes` if users want to see them as floats instead of Fractions.

@jamesturk @drdeford @jgibson517 @jennjwang @ziglaser

I will add these to the change log only once the PR has been merged.

### CODE REVIEW

**Comment by cdonnay** [comment]
Further edits:

1. Fixed error in STV class where if two candidates were elected simultaneously, the fractional transfer would occur twice.
2. Added a tie string to the printing of `ElectionState` so users can see if candidates were tied when elected.

File: src/votekit/election_state.py [hunk 1]
```diff
@@ -91,8 +91,9 @@ def round_outcome(self, round: int) -> dict:
         """
         if self.curr_round == round:
             return {
-                "Elected": [c for s in self.elected for c in s],
-                "Eliminated": [c for s in self.eliminated_cands for c in s],
+                "Elected": [s for s in self.elected ],
```
Comment (jamesturk) [inline src/votekit/election_state.py:94]: nit: I'd probably use list(self.elected) here for clarity and a tiny speed improvement

File: src/votekit/pref_profile.py [hunk 2]
```diff
@@ -261,28 +265,24 @@ def condense_ballots(self):
...
         if not isinstance(other, PreferenceProfile):
             return False
-        self.condense_ballots()
-        other.condense_ballots()
-        for b in self.ballots:
-            if b not in other.ballots:
-                return False
-        for b in self.ballots:
-            if b not in other.ballots:
+        pp_1 = self.condense_ballots()
```
Comment (jamesturk) [inline src/votekit/pref_profile.py:282]: does this change neglect case where pp_2 has a ballot not in pp_1?

**Review by jamesturk — COMMENTED** [review]
Mostly looks good, flagged one small thing and one potential error if I'm not misunderstanding context.

I'd also recommend continuing to run black/ruff on this, as there are quite a few deviations from PEP8 style that they'll pick up.

**Comment by cdonnay** [comment]
@jamesturk  thanks for the edits!

I'm currently running 
- poetry run mypy src
- poetry run ruff src tests
- poetry run pytest
Is there another one I should do?

File: src/votekit/graphs/pairwise_comparison_graph.py [hunk 3]
```diff
@@ -216,3 +232,32 @@ def dominating_tiers(self) -> list[set]:
...
+        Returns a list of condorcet cycles in the graph, which we define as any cycle of length
+            greater than 2.
+
+        Returns:
+            List of condorcet cycles
+        """
+
+        G = self.pairwise_graph
+
+        return(nx.recursive_simple_cycles(G))
```
Comment (drdeford) [inline src/votekit/graphs/pairwise_comparison_graph.py:263]: Do we want to return these sorted by e.g. length?

Comment (cdonnay) [reply src/votekit/graphs/pairwise_comparison_graph.py:263]: I think that's a great idea.

**Pull request merged**